src/avformat/avformat.rs

```diff
@@ -68,58 +81,43 @@ impl AVFormatContextInput {
         // Here we can be sure that context is non null, constructing here for
         // dropping when `avformat_find_stream_info` fails.
         let mut context = unsafe { Self::from_raw(NonNull::new(input_format_context).unwrap()) };
+        context.io_context = io_context;
 
         unsafe { ffi::avformat_find_stream_info(context.as_mut_ptr(), ptr::null_mut()) }
             .upgrade()
             .map_err(RsmpegError::FindStreamInfoError)?;
 
         Ok(context)
     }
+}
 
-    /// Create a [`AVFormatContextInput`] instance from an [`AVIOContext`], and find info of
+impl AVFormatContextInput {
+    /// Create a [`AVFormatContextInput`] instance of a file, and find info of
     /// all streams.
-    pub fn from_io_context(mut io_context: AVIOContextContainer) -> Result<Self> {
-        let input_format_context = {
-            // Only fails on no memory, so unwrap().
-            // `avformat_open_input`'s documentation:
-            //
-            // Note that a user-supplied AVFormatContext will be freed on failure.
-            //
-            // So here we don't construct the `AVFormatContext`, or the
-            // `input_format_context` will be double free.
-            let input_format_context = unsafe { ffi::avformat_alloc_context() }.upgrade().unwrap();
-            unsafe {
-                (*input_format_context.as_ptr()).pb = match &mut io_context {
-                    AVIOContextContainer::Url(ctx) => ctx.as_mut_ptr(),
-                    AVIOContextContainer::Custom(ctx) => ctx.as_mut_ptr(),
-                };
-            }
-            input_format_context
-        };
-
-        unsafe {
-            ffi::avformat_open_input(
-                &mut input_format_context.as_ptr(),
-                ptr::null_mut(),
-                ptr::null_mut(),
-                ptr::null_mut(),
-            )
-        }
-        .upgrade()
-        .map_err(RsmpegError::OpenInputError)?;
-
-        // After `avformat_open_input`, we can `avformat_close_input` after it,
-        // so here we can safely construct a `AVFormatContextInput`.
-        let mut input_format_context = unsafe { Self::from_raw(input_format_context) };
-        input_format_context.io_context = Some(io_context);
-
-        unsafe {
-            ffi::avformat_find_stream_info(input_format_context.as_mut_ptr(), ptr::null_mut())
-        }
-        .upgrade()
-        .map_err(RsmpegError::FindStreamInfoError)?;
+    ///
+    /// - `url`: url of the stream to open.
+    /// - `format`: input format hint. If `format` is some, this parameter forces a specific input format.
+    /// - `options`: A dictionary filled with AVFormatContext and demuxer-private options.
+    ///    On return this parameter will be destroyed and replaced with a dict containing
+    ///    options that were not found.
+    #[deprecated = "see `AVFormatContextInput::builder`"]
+    pub fn open(
```

Comment:
Don't mark this as deprecated as this can be a shortcut. Instead, remove the `fmt` and `options` parameters as they usually set to `None`.